tests/unit_tests/test_capi.py

```diff
@@ -311,3 +311,31 @@ def test_mesh(capi_init):
 
     msf = openmc.capi.MeshSurfaceFilter(mesh)
     assert msf.mesh == mesh
+
+
+def test_restart(pincell_model):
```

Comment:
The problem you are seeing is probably because `capi_init` is a module-level fixture which means that the call to `finalize()` there won't happen until all tests in the module are done. What you might want to do is change this test to use the `capi_init` fixture, call `hard_reset()` at the beginning instead of `init()` and then don't call `finalize()` at the end (i.e., a second time).

Another aside- we probably ought to prevent segfaults when `openmc_init` is called twice.